spec/models/shipping_location_spec.rb

```diff
@@ -2,11 +2,48 @@
 require 'shipping_location'
 
 RSpec.describe ShippingLocation do
+  let(:valid_loc) { ShippingLocation.new(city: "Beaverton", state: "OR", zip: 97005, country: "USA")}
+  let(:invalid_loc) { ShippingLocation.new }
   describe "validations" do
-    it "requires a city" do
-      s = ShippingLocation.new
-      expect(s).to_not be_valid
-      expect(s.errors.keys).to include :city
+    context "invalid" do
+      it "requires a city" do
+        expect(invalid_loc).to_not be_valid
+        expect(invalid_loc.errors.keys).to include :city
+      end
+
+      it "requires a state/province" do
+        expect(invalid_loc).to_not be_valid
+        expect(invalid_loc.errors.keys).to include :province
+      end
+
+      it "requires a zip/postal code" do
+        expect(invalid_loc).to_not be_valid
+        expect(invalid_loc.errors.keys).to include :postal_code
+      end
+
+      it "requires a country" do
+        expect(invalid_loc).to_not be_valid
+        expect(invalid_loc.errors.keys).to include :country
+      end
+    end
+
+    context "valid" do
+      it "should be a valid instance" do
+        expect(valid_loc).to be_valid
+        expect(valid_loc).to be_an_instance_of ShippingLocation
+      end
+    end
+
+    context "inheritance" do
```

Comment:
I'm not sure that I understand why this is in the "validations" block -- could you explain it to me?